src/ui.rs

```diff
@@ -126,17 +130,16 @@ pub fn process_normal_mode<'a>(state: &mut State<'a>, event: &Event) -> Action<'
                 state.render_infobar = !state.render_infobar;
                 state.process_action(Action::ReRender)
             }
-            W | PageUp => Action::SkipForward,
-            B | PageDown => Action::SkipBack,
-            Z => Action::ToggleFit,
+            W | PageUp => state.process_action(Action::SkipForward),
+            B | PageDown => state.process_action(Action::SkipBack),
+            Z => state.process_action(Action::ToggleFit),
             Home => Action::First,
```

Comment:
Please process the action for all the actions. Otherwise if some first deletes an image, then goes to the first image, then uses the dot operator, they will delete another image, and they're probably not expecting that.